asdf_astropy/converters/unit/quantity.py

```diff
@@ -39,4 +45,6 @@ def from_yaml_tree(self, node, tag, ctx):
             value = value._make_array()
             dtype = value.dtype
 
-        return Quantity(value, unit=node["unit"], copy=copy, dtype=dtype)
+        class_ = import_masked_type("Quantity") if isinstance(value, np.ma.MaskedArray) else Quantity
```

Comment:
```suggestion
        if isinstance(value, np.ma.MaskedArray):
            if not minversion("astropy", "7.1.dev"):    # ASTROPY_LT_7_1
                msg = "MaskedQuantity support requires astropy 7.1 or later"
                raise NotImplementedError(msg)
            class_ = Masked(Quantity)
        else:
            class_ = Quantity
```
This requires adding 2 more imports around line 36:
```python
from astropy.utils import minversion
from astropy.utils.masked import Masked
```